website/docs/guides/version_4_upgrade.html.markdown

```diff
@@ -0,0 +1,144 @@
+---
+layout: "vault"
+page_title: "Terraform Vault Provider 4.0.0 Upgrade Guide"
+sidebar_current: "docs-vault-provider-version-3-upgrade"
+description: |-
+  Terraform Vault Provider 4.0.0 Upgrade Guide
+
+---
+
+# Terraform Vault Provider 4.0.0 Upgrade Guide
+
+Version `4.0.0` of the Vault provider for Terraform is a major release and
+includes some changes that you will need to consider when upgrading. This guide
+is intended to help with that process and focuses only on the changes necessary
+to upgrade from version `3.25.0` to `4.0.0`.
+
+Most of the changes outlined in this guide have been previously marked as
+deprecated in the Terraform `plan`/`apply` output throughout previous provider
+releases, up to and including 3.25.0. These changes, such as deprecation notices,
+can always be found in the [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md).
+
+-> If you are upgrading from `1.9.x`. Please follow the
+[2.0.0 Upgrade Guide](./version_2_upgrade.html) before proceeding any further.
+
+-> If you are upgrading from `2.24.x`. Please follow the
+[3.0.0 Upgrade Guide](./version_3_upgrade.html) before proceeding any further.
+
+## Why version 4.0.0?
+
+We introduced version `4.0.0` of the Vault provider in order to make
+performance improvements for deployments that manage many Vault secret or auth
+engine mounts. This improvement required changes to the underlying Vault API
+calls, which in turn would require policy adjustments in environments where
+permissions are least privilege.
+
+The change was deemed significant enough to warrant the major version bump.
+In addition to the aforementioned performance improvements, all previously deprecated fields
+and resources have been removed.
+
+While you may see some small changes in your configurations as a result of
+these changes, we don't expect you'll need to make any major refactorings.
+However, please pay special attention to the changes noted in the TODO(JM) section.
+
+## Which Terraform versions are supported?
+
+Terraform versions `0.12.x` and greater are fully supported. Support for `0.11.x` has been removed.
+If you are still on one of the `0.11.x` versions we recommend upgrading to the latest stable release of Terraform.
+
+Please see the [Terraform Upgrade Guide](https://www.terraform.io/upgrade-guides/index.html)
+for more info about upgrading Terraform.
+
+## I accidentally upgraded to 4.0.0, how do I downgrade to `2.X`?
```

Comment:
Is there a reason why we're providing a downgrade guide to 2.x but not 3.x?

Reply:
mistake, fixed